### PR TITLE
xbps-src: pre-pkg: allow only white-listed directories.

### PR DESCRIPTION
**Pull request by sgn**

### CODE REVIEW

**Comment by sgn** [comment]
Hi @void-linux/pkg-committers 

Can I have your opinions on this change to `xbps-src`?
If no-one complains, I'll push it this weekend.

**Comment by Chocimier** [comment]
`hiawatha` explicitly uses /srv

**Comment by sgn** [comment]
I tempted to fixup the second patch into the first one.

**Comment by jnbr** [comment]
`INSTALL.msg` and `REMOVE.msg` are missing

**Comment by q66** [comment]
why?

also toplevel libdirs are not allowed, we only have `usr/lib` except symlinks in `base-files`

and `lib64` is never allowed in general

**Comment by sgn** [comment]
> why?

To avoid mistake likes messed configure variable


> also toplevel libdirs are not allowed, we only have `usr/lib` except symlinks in `base-files`

It's checked by other code in the top of that file,
`base-files` will provide the symlink in the top-level
https://github.com/void-linux/void-packages/blob/master/common/hooks/pre-pkg/99-pkglint.sh#L10

> and `lib64` is never allowed in general

File: common/hooks/pre-pkg/99-pkglint.sh [hunk 1]
```diff
@@ -21,13 +21,40 @@ hook() {
...
 	done
 
+	for f in "$PKGDESTDIR"/*; do
+		f="${f##*/}"
+		case "$f" in
+		'*')	# The filename is exactly '*'
+			if [ -f "${PKGDESTDIR}/*" ]; then
+				msg_red "${pkgver}: File /${f} is not allowed\n"
+				error=1
+			elif [ "$build_style" != "meta" ]; then
```
Comment (Chocimier) [inline common/hooks/pre-pkg/99-pkglint.sh:38]: It passes package not being `meta` containing just INSTALL.msg with this change.

Comment (sgn) [reply common/hooks/pre-pkg/99-pkglint.sh:38]: I think I'll just drop this `'*'` case.
`xbps-src` will always create `INSTALL` file, after all?

Or is it better to keep it, add a variable `nonmetafile`, set it to 1 in `lib|...`?
using find later for checking empty package is not the best solution, I think

**Comment by Chocimier** [comment]
lgtm

File: common/hooks/pre-pkg/99-pkglint.sh [hunk 2]
```diff
@@ -21,13 +22,44 @@ hook() {
...
+			;;
+		*)
+			msg_red "${pkgver}: /${f} directory is not allowed, remove it!\n"
+			error=1
+			;;
+		esac
+	done
+
+	if [ "$build_style" != meta -a "$emptypkg" = yes ]; then
+		# Forbid empty packages unless build_style=meta
```
Comment (Piraty) [inline common/hooks/pre-pkg/99-pkglint.sh:58]: put the comment above `if`?

Comment (sgn) [reply common/hooks/pre-pkg/99-pkglint.sh]: Yay, fixed.

**Pull request merged**